diff --git a/drivers/media/test-drivers/vivid/vivid-cec.c b/drivers/media/test-drivers/vivid/vivid-cec.c
index 1f7469ff04d5..216340b293f3 100644
--- a/drivers/media/test-drivers/vivid/vivid-cec.c
+++ b/drivers/media/test-drivers/vivid/vivid-cec.c
@@ -203,7 +203,7 @@ int vivid_cec_bus_thread(void *_dev)
 		 * have, then compensate by adjusting the wait time needed for the
 		 * bus to be signal-free for 3 bit periods (the retry time).
 		 */
-		delta_us = div_s64(end - start, 1000);
+		delta_us = div_s64(ktime_to_ns(end) - ktime_to_ns(start), 1000);
 		delta_us -= wait_xfer_us + wait_arb_lost_us;
 		retry_us = CEC_SIGNAL_FREE_TIME_RETRY * CEC_DATA_BIT_US - delta_us;
 		if (retry_us > CEC_MARGIN_US)
