diff --git a/include/media/media-entity.h b/include/media/media-entity.h
index fea489f03d57..09737b47881f 100644
--- a/include/media/media-entity.h
+++ b/include/media/media-entity.h
@@ -13,11 +13,10 @@
 
 #include <linux/bitmap.h>
 #include <linux/bug.h>
-#include <linux/container_of.h>
 #include <linux/fwnode.h>
+#include <linux/kernel.h>
 #include <linux/list.h>
 #include <linux/media.h>
-#include <linux/types.h>
 
 /* Enums used internally at the media controller to represent graphs */
 
