diff --git a/drivers/media/platform/atmel/atmel-isc.h b/drivers/media/platform/atmel/atmel-isc.h
index 07fa6dbf8460..eefc40331aa1 100644
--- a/drivers/media/platform/atmel/atmel-isc.h
+++ b/drivers/media/platform/atmel/atmel-isc.h
@@ -11,6 +11,8 @@
 #ifndef _ATMEL_ISC_H_
 
 #include <linux/clk-provider.h>
+#include <linux/delay.h>
+#include <linux/irqreturn.h>
 #include <linux/platform_device.h>
 
 #include <media/v4l2-ctrls.h>
