diff --git a/drivers/media/dvb-frontends/si2183.c b/drivers/media/dvb-frontends/si2183.c
index 4aa0b7876340..3f62b1372c62 100644
--- a/drivers/media/dvb-frontends/si2183.c
+++ b/drivers/media/dvb-frontends/si2183.c
@@ -1528,9 +1528,9 @@ static struct si_base *match_base(struct i2c_adapter *i2c, u8 adr)
 
 
 
-static int si2183_probe(struct i2c_client *client)
+static int si2183_probe(struct i2c_client *client,
+						const struct i2c_device_id *id)
 {
-	const struct i2c_device_id *id = i2c_client_get_device_id(client);
 	struct si2183_config *config = client->dev.platform_data;
 	struct si2183_dev *dev;
 	struct si_base *base;
@@ -1664,7 +1664,7 @@ static struct i2c_driver si2183_driver = {
 	.driver = {
 		.name	= "si2183",
 	},
-	.probe_new	= si2183_probe,
+	.probe		= si2183_probe,
 	.remove		= si2183_remove,
 	.id_table	= si2183_id_table,
 };
