diff --git a/drivers/media/radio/radio-shark.c b/drivers/media/radio/radio-shark.c
index 127a3be0e0f0..8230da828d0e 100644
--- a/drivers/media/radio/radio-shark.c
+++ b/drivers/media/radio/radio-shark.c
@@ -316,16 +316,6 @@ static int usb_shark_probe(struct usb_interface *intf,
 {
 	struct shark_device *shark;
 	int retval = -ENOMEM;
-	static const u8 ep_addresses[] = {
-		SHARK_IN_EP | USB_DIR_IN,
-		SHARK_OUT_EP | USB_DIR_OUT,
-		0};
-
-	/* Are the expected endpoints present? */
-	if (!usb_check_int_endpoints(intf, ep_addresses)) {
-		dev_err(&intf->dev, "Invalid radioSHARK device\n");
-		return -EINVAL;
-	}
 
 	shark = kzalloc(sizeof(struct shark_device), GFP_KERNEL);
 	if (!shark)
diff --git a/drivers/media/radio/radio-shark2.c b/drivers/media/radio/radio-shark2.c
index f1c5c0a6a335..d150f12382c6 100644
--- a/drivers/media/radio/radio-shark2.c
+++ b/drivers/media/radio/radio-shark2.c
@@ -282,16 +282,6 @@ static int usb_shark_probe(struct usb_interface *intf,
 {
 	struct shark_device *shark;
 	int retval = -ENOMEM;
-	static const u8 ep_addresses[] = {
-		SHARK_IN_EP | USB_DIR_IN,
-		SHARK_OUT_EP | USB_DIR_OUT,
-		0};
-
-	/* Are the expected endpoints present? */
-	if (!usb_check_int_endpoints(intf, ep_addresses)) {
-		dev_err(&intf->dev, "Invalid radioSHARK2 device\n");
-		return -EINVAL;
-	}
 
 	shark = kzalloc(sizeof(struct shark_device), GFP_KERNEL);
 	if (!shark)
